analytics/forecasting_v2/episode_lifecycle.py

```
@@ -2161,6 +2414,40 @@ def _assert_decision_matches_history(
+        raise V2EpisodeLifecycleError(
+            f"this decision's §18.1 decision_tick_size={risk.decision_tick_size} is not episode "
+            f"{history.episode_id!r}'s own persisted creation value {tick} -- §12.5a freezes the "
+            "creation-time tick grid for the episode's whole life, and today's instrument "
+            "metadata is never consulted")
+    expected = _applicable_invalidation_price(history, decision.operational_facts)
+    if _exact(risk.invalidation_price, "invalidation_price") != _exact(expected, "expected"):
+        raise V2EpisodeLifecycleError(
+            f"this decision's §18.1 invalidation_price={risk.invalidation_price!r} is not the "
+            f"level episode {history.episode_id!r} actually carries at this boundary "
+            f"({expected!r}) -- planned risk must be measured against the frozen geometry the "
+            "episode will really be confirmed with, never a stale or foreign one")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)